src/main/java/io/divolte/server/recordmapping/DslRecordMapping.java

```diff
@@ -967,31 +964,10 @@ public BooleanValueProducer satelliteProvider() {
         }
     }
 
-    /*
-     * Handy when using something other than the remote host as IP address (e.g. custom headers set by a load balancer).
-     * We force the address to be in the numeric IPv4 scheme, to prevent name resolution and IPv6.
-     */
-    private static Optional<InetAddress> tryParseIpv4(final String ip) {
-        final byte[] result = new byte[4];
-        final String[] parts = StringUtils.split(ip, '.');
-        if (parts.length != 4) {
-            return Optional.empty();
-        }
-        try {
-            for (int c = 0; c < 4; c++) {
-                final int x = Integer.parseInt(parts[c]);
-                if (x < 0x00 || x > 0xff) {
-                    return Optional.empty();
-                }
-                result[c] = (byte) (x & 0xff);
-            }
-        } catch(final NumberFormatException nfe) {
-            return Optional.empty();
-        }
-
+    private static Optional<InetAddress> tryParseIp(final String ip) {
```

Comment:
I have no recollection of why we previously forced this to be IPv4. Cleary we aimed to avoid name resolution, which is what the Guava method does as well according to its docs. I think the likely explanation for forcing IPv4 previously is that it was easier to implement that way.

Reply:
That's basically my take on it as well.